frontend/src/app/api/BaseApi.ts

```diff
@@ -15,32 +15,26 @@ import {
   ValidationError,
 } from "src/errors";
 import { QueryParamData } from "src/services/search/searchfetcher/SearchFetcher";
-// TODO (#1682): replace search specific references (since this is a generic API file that any
-// future page or different namespace could use)
 import { APIResponse } from "src/types/apiResponseTypes";
 
 export type ApiMethod = "DELETE" | "GET" | "PATCH" | "POST" | "PUT";
 export interface JSONRequestBody {
   [key: string]: unknown;
 }
 
-interface APIResponseError {
-  field: string;
-  message: string;
-  type: string;
-}
-
 export interface HeadersDict {
   [header: string]: string;
 }
 
 export default abstract class BaseApi {
-  // Root path of API resource without leading slash.
-  abstract get basePath(): string;
+  // Root path of API resource without leading slash, can be overridden by implementing API classes as necessary
+  get basePath() {
```

Comment:
since we are using the same basePath everywhere, I figured we may as well define it at a higher level